test/jdk/javax/swing/JComboBox/8019180/Test8019180.java

```diff
@@ -37,32 +37,44 @@
 public class Test8019180 implements Runnable {
     private static final CountDownLatch LATCH = new CountDownLatch(1);
     private static final String[] ITEMS = {"First", "Second", "Third", "Fourth"};
+    private static JFrame frame;
 
-    public static void main(String[] args) throws InterruptedException {
-        SwingUtilities.invokeLater(new Test8019180());
-        LATCH.await();
+    public static void main(String[] args) throws Exception {
+        try {
+            SwingUtilities.invokeLater(new Test8019180());
+            LATCH.await();
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
     }
 
     private JComboBox<String> test;
 
     @Override
     public void run() {
-        if (this.test == null) {
-            this.test = new JComboBox<>(ITEMS);
-            this.test.addActionListener(this.test);
-            JFrame frame = new JFrame();
-            frame.add(test);
-            frame.pack();
-            frame.setVisible(true);
-            SwingUtilities.invokeLater(this);
-        } else {
-            int index = this.test.getSelectedIndex();
-            this.test.setSelectedIndex(1 + index);
-            if (0 > this.test.getSelectedIndex()) {
-                System.err.println("ERROR: no selection");
-                System.exit(8019180);
+        try {
+            if (this.test == null) {
+                this.test = new JComboBox<>(ITEMS);
+                this.test.addActionListener(this.test);
+                frame = new JFrame();
+                frame.add(test);
+                frame.setLocationRelativeTo(null);
+                frame.pack();
+                frame.setVisible(true);
+                SwingUtilities.invokeLater(this);
+            } else {
+                int index = this.test.getSelectedIndex();
+                this.test.setSelectedIndex(1 + index);
+                if (0 > this.test.getSelectedIndex()) {
+                    System.err.println("ERROR: no selection");
+                    throw new RuntimeException("Combobox not selected");
```

Comment:
So this is relying on jtreg to catch exceptions not handled by a thread ?
Wouldn't it be better to flag the failure and have it thrown by the main thread ?

BTW since all the client tests are run in othervm mode, System.exit() isn't a real problem but it is still fine to fix this.

Reply:
modified to throw exception in main